lib/connect.c

```diff
@@ -127,7 +127,7 @@ tcpkeepalive(struct SessionHandle *data,
     if(WSAIoctl(sockfd, SIO_KEEPALIVE_VALS, (LPVOID) &vals, sizeof(vals),
                 NULL, 0, &dummy, NULL, NULL) != 0) {
       infof(data, "Failed to set SIO_KEEPALIVE_VALS on fd %d: %d\n",
-            (int)sockfd, WSAGetLastError());
+            curlx_sktosi(sockfd), WSAGetLastError());
```

Comment:
Looking on `curlx_sktosi()` definition, it should produce less number of compiler warnings.
For `SOCKET` -> `int` conversion simple cast is enough, but reverse conversion should be sign-extending, not simple type casting (correctly implemented as function in warnless.c).
The main reason for using this function was symmetry - both `SOCKET` -> `int` and `int` -> `SOCKET` used as function. Otherwise it's very tempting for non-windows programmer to use simple cast in reverse conversion.